Increase test coverage: Array.prototype.toLocaleString

matedabis opened the pull request:
The coverage is measured by the following script, using `--jerry-test-suite --test262 --unittests --jerry-tests`:
https://github.com/matedabis/jerryscript/blob/gcov_coverage_tester/tests/gcov-tests/gcovtester.py

Branch coverage:
Before: 15/18
After: 18/18

JerryScript-DCO-1.0-Signed-off-by: Mate Dabis mdabis@inf.u-szeged.hu

rerobika commented on tests/jerry/array-prototype-tolocalestring.js:
```diff
@@ -54,3 +54,33 @@ try {
+
+// Checking behavior when null is passed to the function
+try {
+  Array.prototype.toLocaleString.call(null);
```
Add an `assert (false)` after this call.

rerobika commented on tests/jerry/array-prototype-tolocalestring.js:
```diff
@@ -54,3 +54,33 @@ try {
+  Array.prototype.toLocaleString.call(obj);
+  assert(false);
+} catch (e) {
+  assert(e instanceof ReferenceError);
```
Please check the message as well! (And use unique messages)

matedabis replied:
The thrown error and the one in the function are not the same `ReferenceError`s, so as I check the message, I get assertion fail.

akosthekiss replied:
Something is not right then. Q1: Why set the error message then, if it cannot be checked? Q2: Does the error object get replaced somewhere by the builtin implementation? Q3: If yes, why? Q4: If yes, is it necessary to throw a reference error within the anonymous function or any kind of error can be thrown?

matedabis replied:
Sorry I probably made a typo, thats why it did not work. It works with the message check.

matedabis replied:
Sorry I made a typo, that is why it did not work.

rerobika commented on tests/jerry/array-prototype-tolocalestring.js:
```diff
@@ -54,3 +54,33 @@ try {
+  assert(e instanceof ReferenceError);
+}
+
+// Checking behavior when length is an object, wichs valueOf throws error
```
typo: `wichs`

matedabis commented:
Corrected according to the reviews.
Replying to this comment: https://github.com/pando-project/jerryscript/pull/2716#discussion_r249092528
in the get function, and which is being catched are not the same `ReferenceError`s, as I check the message, I get assertion fail.

rerobika commented on tests/jerry/array-prototype-tolocalestring.js:
```diff
@@ -54,3 +54,34 @@ try {
   assert (e instanceof ReferenceError);
 }
+
+// Checking behavior when null is passed to the function
```
This description is a bit misleading. I'd suggest the same as in #2717.

matedabis commented:
I updated the patch according to the review.

matedabis commented:
Updated the patch:
- Added comments for each case containing which part of the ES 5.1 it tests.
- Ran the test code in Gecko, V8, SpiderMonkey engines and got the same result.

matedabis commented:
Corrected the issue above.

matedabis commented:
Closing PR because of moving to another branch.
